models/yiiModels/DataAnalysisAppSearch.php

```diff
@@ -0,0 +1,232 @@
+                $visualisationsInfo[self::APP_SHORT_NAME] = explode("/", $app)[1] . "-" . $functionName;
+                $visualisationsInfo[self::R_PACKAGE_NAME] = $app;
+                $url = "$applicationWebPath/index.html?accessToken=" . Yii::$app->session[WSConstants::ACCESS_TOKEN] . "&wsUrl=" . WS_PHIS_PATH;
+                $visualisationsInfo[self::APP_INDEX_HREF] = Url::to(['data-analysis/view', 'url' => $url, 'name' => explode("/", $app)[1]]);
+            }
+        }
+        return $visualisationsInfo;
+    }
+
+    /**
+     * Return exported functionalities 
+     * @param string $applicationName 
+     * @return array functionalities
+     */
+    function getAppFunctionalities($applicationName) {
+         if ($this->ocpuserver->status()) {
+            try {
+                $response = $this->ocpuserver->getOpenCPUWebServerClient()->request(OpenCPUServer::OPENCPU_SERVER_GET_METHOD, 'apps/' . $applicationName . '/opensilex/descriptions');
+                $body = $response->getBody();
+                // retrevies body as a string
+                $stringBody = (string) $body;
+                $sessionValuesResults = explode("\n", $stringBody);
+                $cleansessionValuesResults = array_filter($sessionValuesResults);
+
+                return $cleansessionValuesResults;
+            } catch (RequestException $e) {
+                $errorMessage = Psr7\str($e->getRequest());
+                if ($e->hasResponse()) {
+                    $errorMessage .= '--' . Psr7\str($e->getResponse());
+                }
+                $this->serverCallStatus = new CallStatus($errorMessage, $e->getResponse()->getStatusCode(), $e);
+                // ClientException is thrown for 400 level errors
+            } catch (ClientException $e) {
+                $errorMessage = Psr7\str($e->getRequest());
+                if ($e->hasResponse()) {
+                    $errorMessage .= '--' . Psr7\str($e->getResponse());
+                }
+                $this->serverCallStatus = new CallStatus($errorMessage, $e->getResponse()->getStatusCode(), $e);
+                // is thrown for 500 level errors
+            } catch (ServerException $e) {
+                $errorMessage = Psr7\str($e->getRequest());
+                if ($e->hasResponse()) {
+                    $errorMessage .= '--' . Psr7\str($e->getResponse());
+                }
+                $this->serverCallStatus = new CallStatus($errorMessage, $e->getResponse()->getStatusCode(), $e);
+            }
+        }
+        return [];
+    }
+
+  
+     /**
+     * Return R application configuration
+     * @param string $app
+     * @return array
+     */
+    function getAppConfiguration($app) {
+        if ($this->ocpuserver->status()) {
+            try {
+                $response = $this->ocpuserver->getOpenCPUWebServerClient()->request(OpenCPUServer::OPENCPU_SERVER_GET_METHOD, 'apps/' . $app . '/opensilex/webAppConfig.yml');
```

Comment:
Add explanation

Reply:
I add some application R structure comments to this function.